build_single.sh

```
@@ -0,0 +1,150 @@
+    echo "$REGISTRY/bun-node:${NODE_MAJOR}-${distro}"
+  fi
+}
+
+bun_build_arg="$BUN_VERSION"
+if [[ "$BUN_VERSION" == *"-canary"* ]]; then
+  bun_build_arg="canary"
+fi
+
+log "Building image for Bun version $BUN_VERSION, Node version $NODE_VERSION, Distro $DISTRO"
+image_name="$REGISTRY/bun-node:${BUN_VERSION}-${NODE_VERSION}-${tag_distro}"
+tags=($(generate_tags "$BUN_VERSION" "$NODE_VERSION" "$tag_distro"))
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)